contracts/hyphen/LiquidityPool.sol

```diff
@@ -359,7 +345,15 @@ contract LiquidityPool is
 
             uint256 length = updatedPayloads.length;
             for (uint256 i = 1; i < length; ) {
-                updatedPayloads[i] = args.payloads[i - 1];
+                // Other payloads must not call the liquidity pool contract on destination chain,
+                // else an attacker can simply ask the liquidity pool to send all funds on the exit chain
+                ICCMPGateway.CCMPMessagePayload memory payload = args.payloads[i - 1];
+                require(payload.to != toChainLiquidityPoolAddress, "51");
```

Comment:
Is payload.to the only way to call LiquidityPool on other chain, what if someone build a calldata that internally calls the LiquidityPool may be via delegateCall?

Reply:
There are two cases here:
1. Internal `call`: Will fail since msg.sender != ccmpExecutor.
2. Internal `delegatecall`: This would then update state of the contract calling the liquidity pool, not the liquidity pool itself. Therefore, user funds are safe.